src/catch2/internal/catch_run_context.cpp

```diff
@@ -450,6 +450,13 @@ namespace Catch {
         assertionEnded(CATCH_MOVE(result) );
         resetAssertionInfo();
 
+        // Best effort cleanup for sections that have not been destructed yet
+        // Since this is a fatal error, we have not had and won't have the opportunity to destruct them properly
+        while (!m_activeSections.empty()) {
+            auto nl = m_activeSections.back()->nameAndLocation();
+            SectionEndInfo endInfo{ SectionInfo(CATCH_MOVE(nl.location), CATCH_MOVE(nl.name)), {}, 0.0 };
+            sectionEndedEarly(CATCH_MOVE(endInfo));
+        }
```

Comment:
I added a test that confirms that with this change JUnit does not hard crash when handling a signal.

I'll have to think about the change a bit more before merging them though.

Reply:
Yeah, this seems fine. Not moving the nameAndLocation is probably better. It _should_ be fine, but also something that could easily cause issues in the future, after some refactoring.